src/engine/bufferscalers/enginebufferscalerubberband.cpp

```diff
@@ -140,17 +140,29 @@ void EngineBufferScaleRubberBand::clear() {
 SINT EngineBufferScaleRubberBand::retrieveAndDeinterleave(
         CSAMPLE* pBuffer,
         SINT frames) {
-    SINT frames_available = m_pRubberBand->available();
-    SINT frames_to_read = math_min(frames_available, frames);
+    const SINT frames_available = m_pRubberBand->available();
+    const SINT frames_to_read = math_min(frames_available, frames);
     SINT received_frames = static_cast<SINT>(m_pRubberBand->retrieve(
             m_bufferPtrs.data(), frames_to_read));
```

Comment:
Fixed.

Reply:
No, did you miss to push a commit? 

The function expects `frames` and we need to discard `frame_offset` from the retrieved buffer. So we need to retrieve `frames +  frame_offset`.

Reply:
See 755a4a485c81376228a9f4a09d35759f2e9a5626.

Reply:
The linked code makes that the function returns less frames than it might could have read. So we should not clamp the value to frames , but to frames + frame_offset in line 132

Reply:
No we should not. I can't check what `m_pRubberBand->available()` returns here, but logically it would return 1024 if it's doing overlap-add with four times overlap. At a 1.0 pitch ratio/when the time stretcher should not be making any adjustments, we need to throw away half a window's worth of samples, which is 2048 in this case. So in that situation the first two calls to this function _must_ read and return 0 samples or else we've been reading the processed padding that we're trying to throw away here.

> The linked code makes that the function returns less frames than it might could have read.

It can only read a small window of output at a time. Presumably this is 1/4th of the FFT window size. So 1024 samples. Which means that the first two calls to this function (each producing 1024 samples of garbage output) should be thrown away.

Reply:
I added a bit debug code to shows the issue after a seek: 
`    qDebug() << "padding  = " << m_remainingPaddingInOutput << "frames  = " << frames << "available  = " << frames_available;`

```
debug [Engine] padding  =  1025 frames  =  256 available  =  0
debug [Engine] padding  =  1025 frames  =  256 available  =  512
debug [Engine] padding  =  769 frames  =  256 available  =  768                 
debug [Engine] padding  =  513 frames  =  256 available  =  1024
debug [Engine] padding  =  257 frames  =  256 available  =  1280
debug [Engine] padding  =  1 frames  =  256 available  =  1536
debug [Engine] padding  =  0 frames  =  1 available  =  1792
debug [Engine] padding  =  0 frames  =  256 available  =  1791
debug [Engine] padding  =  0 frames  =  256 available  =  1535
debug [Engine] padding  =  0 frames  =  256 available  =  1279
```
In the second call, we read only 256 frames from the available 512. The available frames are stacked up before they are reduced again. We need to read all available frames in that case. This needs to be fixed.

Reply:
Oh yeah I see what you mean now! Fixed in d3aa2104c083ba6d0a540d31f8b03560fafa70f2.

Somewhat related, would there be any interest in exposing the Rubberband R3 + short window option? I added it for myself here: https://github.com/robbert-vdh/mixxx/commit/af570f481394bcc3eab399a570a5e25b406ac0ba Its performance is in line with R2 (and also no huge spikes when searching), and quality wise it's more or less a sidegrade to R2. It doesn't have R2's time domain transient preservation, but it also doesn't introduce spurious transients and it does still sound better than R2 without transient preservation.